sci-wcalc/coax_calc.man

.\" Copyright (c), 2005 Dan McMahill
.\" Do not edit this directly.  Edit the XML source file instead
.\"

.TH coax_calc "" "" "Wcalc" "Wcalc Commands"
.B coax_calc
- Coaxial transmission line analysis

.SH SYNOPSIS

[L, R, C, G, lc, ld, fc, em] = 
coax_calc(a, b, c, t, rho_a, rho_b, er, tand, len, f)


.SH INPUT PARAMETERS

.TP 10
a
: radius of inner conductor (meters)
.TP 10
b
: inner radius of outer conductor (meters)
.TP 10
c
: offset of the inner conductor from the center (meters)
.TP 10
t
: thickness of shield (meters)
.TP 10
rho_a
: resistivity of inner conductor (ohms/meter)
.TP 10
rho_b
: resistivity of outer conductor (ohms/meter)
.TP 10
er
: relative permitivitty of dielectric
.TP 10
tand
: loss tangent of dielectric
.TP 10
len
: physical length of line (meters)
.TP 10
f
: analysis frequency (Hertz)

.SH OUTPUT VALUES

.TP 10
L
: calculated inductance per length (Henries/meter)
.TP 10
R
: calculated resistance per length (Ohms/meter)
.TP 10
C
: calculated capacitance per length (Farads/meter)
.TP 10
G
: calculated conductance per length (Siemens/meter)
.TP 10
lc
: conductor loss (dB/meter)
.TP 10
ld
: dielectric loss (dB/meter)
.TP 10
fc
: TE10 cutoff frequency (Hz)
.TP 10
em
: Max Electric Field (Volts/meter)
.SH DESCRIPTION

Function to analyze the electrical characteristics of a
coaxial transmission line.
The physical
dimensions of the line are given as inputs and the 
characteristic impedance and incremental circuit model
elements are calculated.

.SH EXAMPLE
.nf

.TP
 // mils to meters conversion factor
sf=25.4e-6;
.TP
 // radius of inner conductor
a=10.0*sf;
.TP
 // inner radius of outer conductor
b=100.0*sf;
.TP
 // offset of inner conductor
c=0*sf;
.TP
 // thickness of outer conductor
t=5.0*sf;
.TP
 // resistivity of inner conductor
rho_a=3.5e-8;
.TP
 // resistivity of outer conductor
rho_b=rho_a;
.TP
 // relative permitivitty of dielectric
er=9.5;
.TP
 // dielectric loss tangent
tand=0.01;
.TP
 // length of line
len=1.0;
.TP
 // 100 MHz operation
f=100e6;
[z0,elen,loss,L,R,C,G,lc,ld,fc,em] = coax_calc(a,b,c,t,rho_a,rho_b,er,tand,len,f);
disp(sprintf('z0   = %g Ohms',z0));
disp(sprintf('elen = %g degrees',elen));
disp(sprintf('loss = %g dB',loss));
disp(sprintf('L    = %g H/m',L));
disp(sprintf('R    = %g Ohms/m',R));
disp(sprintf('C    = %g F/m',C));
disp(sprintf('G    = %g S/m',G));
disp(sprintf('Conductor loss  = %g dB/m', lc));
disp(sprintf('Dielectric loss = %g dB/m', ld));
disp(sprintf('TE10 cutoff     = %g GHz', fc/1e9));
disp(sprintf('Max E field     = %g kV/m', em/1e3));
.fi
.SH SEE ALSO
stdio-wcalc(1),
air_coil_calc(), air_coil_syn(), coax_calc(), coax_syn(), coplanar_calc(), coplanar_syn(), coupled_microstrip_calc(), coupled_microstrip_syn(), coupled_stripline_calc(), coupled_stripline_syn(), ic_microstrip_calc(), ic_microstrip_syn(), microstrip_calc(), microstrip_syn(), stripline_calc(), stripline_syn()
.SH AUTHOR

Dan McMahill

.SH BUGS

None known


NEWS
$Id: NEWS,v 1.10 2009/02/28 03:26:18 dan Exp $

Wcalc -- history of user-visible changes. 

Copyright (c) 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005
2006, 2007, 2008, 2009 Dan McMahill

Please report bugs on the sourceforge project page.

===============
Version 1.1
===============

Added series/parallel RC equivalent circuit calculator.

Added series/parallel RL equivalent circuit calculator.

Added self and mutual inductance of two rectangular bars.

Added coplanar waveguide model.

Added coupled stripline model.

Corrected the Q calculations for low frequency (not skindepth) region
in air core inductors.  Also corrected Q calculation for conductors
other than copper.

Corrected the calculation of incremental conductance for the
microstrip model.

Corrected a bug in the calculation of conductor losses in the
microstrip model.

Added/enabled code for calculating conductor losses in the coupled
microstrip model.

When compiling with gtk-2.10 or newer, use gtkPrint for printing.
This gives a more professional/standard print dialog and also enables
printing under windows.

Added a preliminary Dutch translation.

Converted the figures to all use solid grayscale fills instead of
pattern fills as the former prints better.

Several updates to the build system to moderize it a bit.  Uses
AM_CONDITIONALS more appropriately, uses AC_PROG_LIBTOOL instead of
AM_PROG_LIBTOOL, etc.

Update the win32 build script to generate an installer that works
correctly under vista.

Add .wc file associations under vista.

Update the octave build infrastructure to work with octave-3.0 and
newer where mex support is built into mkoctfile instead of a
standalone mex shell script.

Switch to GPL licensing.

Create several utility functions to make it faster to build new gtk
forms.  This is only visible to developers.

Improved autogen.sh to check versions of several of the auto* tools to
help make it easier to diagnose any issues which may come up.

Repair compilation with the SunPRO compilers.


===============
Version 1.0
===============
Replaced the alert dialog box in the gtk version with a log
window.  This prevents a problem when many alerts are generated
and the user is presented with many many alert windows.

Added win32 support along with a build script for a windows installer.

Fixed a bug which gave lots of error/warning messages in the cgi
frontend for coupled microstrip.

===============
Version 0.9
===============
First public release.

===============
Version 0.1 
===============
Initial tagged version.  This does good analysis and synthesis
of microstrip.  Considered stable for this purpose.  However,
open/save/print and units other than mils are not supported yet.



octave-wcalc/coax_calc.m
%COAX_CALC        - Coaxial transmission line analysis 
%  
%  SYNOPSIS
%  
%  [L, R, C, G, lc, ld, fc, em] = coax_calc(a, b, c, t, rho_a, rho_b, er, tand,
%  len, f)
%  
%  PARAMETERS
%  
%  INPUT PARAMETERS
%  
%    * a : radius of inner conductor (meters)
%    * b : inner radius of outer conductor (meters)
%    * c : offset of the inner conductor from the center (meters)
%    * t : thickness of shield (meters)
%    * rho_a : resistivity of inner conductor (ohms/meter)
%    * rho_b : resistivity of outer conductor (ohms/meter)
%    * er : relative permitivitty of dielectric
%    * tand : loss tangent of dielectric
%    * len : physical length of line (meters)
%    * f : analysis frequency (Hertz)
%  
%  OUTPUT VALUES
%  
%    * L : calculated inductance per length (Henries/meter)
%    * R : calculated resistance per length (Ohms/meter)
%    * C : calculated capacitance per length (Farads/meter)
%    * G : calculated conductance per length (Siemens/meter)
%    * lc : conductor loss (dB/meter)
%    * ld : dielectric loss (dB/meter)
%    * fc : TE10 cutoff frequency (Hz)
%    * em : Max Electric Field (Volts/meter)
%  
%  DESCRIPTION
%  
%  Function to analyze the electrical characteristics of a coaxial transmission
%  line. The physical dimensions of the line are given as inputs and the
%  characteristic impedance and incremental circuit model elements are calculated.
%  
%  EXAMPLE
%  
%   % mils to meters conversion factor
%  sf=25.4e-6;
%   % radius of inner conductor
%  a=10.0*sf;
%   % inner radius of outer conductor
%  b=100.0*sf;
%   % offset of inner conductor
%  c=0*sf;
%   % thickness of outer conductor
%  t=5.0*sf;
%   % resistivity of inner conductor
%  rho_a=3.5e-8;
%   % resistivity of outer conductor
%  rho_b=rho_a;
%   % relative permitivitty of dielectric
%  er=9.5;
%   % dielectric loss tangent
%  tand=0.01;
%   % length of line
%  len=1.0;
%   % 100 MHz operation
%  f=100e6;
%  [z0,elen,loss,L,R,C,G,lc,ld,fc,em] = coax_calc(a,b,c,t,rho_a,rho_b,er,tand,len,f);
%  disp(sprintf('z0   = %g Ohms',z0));
%  disp(sprintf('elen = %g degrees',elen));
%  disp(sprintf('loss = %g dB',loss));
%  disp(sprintf('L    = %g H/m',L));
%  disp(sprintf('R    = %g Ohms/m',R));
%  disp(sprintf('C    = %g F/m',C));
%  disp(sprintf('G    = %g S/m',G));
%  disp(sprintf('Conductor loss  = %g dB/m', lc));
%  disp(sprintf('Dielectric loss = %g dB/m', ld));
%  disp(sprintf('TE10 cutoff     = %g GHz', fc/1e9));
%  disp(sprintf('Max E field     = %g kV/m', em/1e3));
%  
%  SEE ALSO
%  
%  air_coil_calc
%  air_coil_syn
%  coax_calc
%  bars_calc
%  coax_syn
%  coplanar_calc
%  coplanar_syn
%  coupled_microstrip_calc
%  coupled_microstrip_syn
%  coupled_stripline_calc
%  coupled_stripline_syn
%  ic_microstrip_calc
%  ic_microstrip_syn
%  microstrip_calc
%  microstrip_syn
%  parallel_rc_p2s
%  parallel_rc_s2p
%  parallel_rl_p2s
%  parallel_rl_s2p
%  stripline_calc
%  stripline_syn
%  
%  AUTHOR
%  
%  Dan McMahill
%  
%  BUGS
%  
%  None known
%  
%  Please report any bugs on the Wcalc Sourceforge Project Page
%  


% Copyright (c) 2001-2009 Dan McMahill
% All rights reserved.
%
% This program is free software; you can redistribute it and/or modify
% it under the terms of the GNU General Public License as published by
% the Free Software Foundation; version 2 of the License.
% 
% This program is distributed in the hope that it will be useful,
% but WITHOUT ANY WARRANTY; without even the implied warranty of
% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
% GNU General Public License for more details.
% 
% You should have received a copy of the GNU General Public License
% along with this program; if not, write to the Free Software
% Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
%
